sdk/search/Azure.Search.Documents/src/SearchClient.cs

```diff
@@ -55,10 +52,9 @@ public class SearchClient
         internal ObjectSerializer Serializer { get; }
 
         /// <summary>
-        /// Gets the authenticated <see cref="HttpPipeline"/> used for sending
-        /// requests to the Search Service.
+        /// The HTTP pipeline for sending and receiving REST requests and responses.
         /// </summary>
-        private HttpPipeline Pipeline { get; }
+        public virtual HttpPipeline Pipeline => _pipeline;
```

Comment:
Tip: just making this `public virtual` would've been enough. This is functionally equivalent. But at least it's consistent with the code above.

Reply:
That said, I know @tg-msft prefers this pattern i.e.,

```C#
public virtual HttpPipeline Pipeline { get; }

public SearchClient()
{
  // ...
  Pipeline = options.Build(credential);
}
```

The public API will look exactly the same. Even with just a `get` accessor - which `=>` generates - there's an implicit private `set` accessor though, in practice, the compiler actually just writes the backing field that gets generated.

Reply:
@heaths - I'm following the DPG generated coding pattern - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/appconfiguration/Azure.Data.AppConfiguration/src/Generated/ConfigurationClient.cs#L34-L35
```csharp
private readonly HttpPipeline _pipeline;

 /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
public virtual HttpPipeline Pipeline => _pipeline;

public SearchClient()
{
  // ...
  _pipeline = options.Build(credential);
}
```

Reply:
I don't feel strongly and we can use whatever's already in the PR.

As Heath noted, I personally like `public virtual HttpPipeline Pipeline { get; }` a _tiny_ bit more because [the implicit private setter is only allowed in the .ctor](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/auto-implemented-properties) so the backing value is otherwise immutable after construction.  I don't think anyone would ever accidentally change `_pipeline` though so I'm good with `public virtual HttpPipeline Pipeline => _pipeline;` too.

Reply:
To clarify, it was only a nit for this PR.